ignite/handlers/checkpoint.py

```diff
@@ -353,6 +370,57 @@ def _setup_checkpoint(self) -> dict:
             checkpoint[k] = obj.state_dict()
         return checkpoint
 
+    @staticmethod
+    def setup_filename_pattern(
+        with_prefix: bool = True,
+        with_score: bool = True,
+        with_score_name: bool = True,
+        with_global_step_transform: bool = True,
+    ) -> str:
+        """Helper method to get the default filename pattern for a checkpoint.
+        Args:
+            with_prefix (bool): If True, the `filename_prefix` with an underscore (filename_prefix_) will be added to the filename pattern.
+                By default the `filename_prefix` will not be appended to the filename pattern
+            with_score (bool): If True, this indicates that the score function is not provided, then it will look for ``with_score_name``
+                and ``with_global_step_transform`` to make the required filename pattern. If it is False then this indicates that the score
+                function is provided.
+            with_score_name (bool): If True, this indicates that the score name is not provided, then it will look for 
+                ``with_global_step_transform`` to make the required filename pattern. If it is False then this indicates that the score
+                name is provided.
+            with_global_step_transform (bool): If True, this indicates that there is no global step, then it will make the
+                required filename pattern accordingly. If it is False then this indicates that there is a global step.
+        Example:
+            .. code-block:: python
+
+                from ignite.handlers import Checkpoint
+
+                filename_pattern = Checkpoint.setup_filename_pattern()
+                print(filename_pattern)
+                > "{filename_prefix}_{name}_{score}.{ext}"
+        """
+        filename_pattern = None
+        if with_score and with_score_name and with_global_step_transform:
+            filename_pattern = "{name}_{score}.{ext}"
```

Comment:
is this contre-intuitive too or I do not understand something ?

"if with score and with score name and with global step transform"  -> I would assume that filepattern contains those field.

Reply:
@vfdev-5 we will have the pattern `{name}_{score}.{ext}` if the score function, score name, and global step transform function are not provided.

Reply:
The code is 
```
if with_score and with_score_name and with_global_step_transform:
            filename_pattern = "{name}_{score}.{ext}"
```
- `with_score` means filepattern will be with the value from score function 
- `with_score_name` means filepattern will be with score name
- `with_global_step_transform` means filepattern will be  with global step
so I expect to have `filename_pattern = "{name}_{global_step}_{score_name}={score}.{ext}"`

Reply:
I have updated accordingly @vfdev-5.